sycl/doc/extensions/proposed/sycl_ext_oneapi_bindless_images.asciidoc

```diff
@@ -0,0 +1,1988 @@
+|======================
+|Device descriptor |Description
+|`aspect::ext_oneapi_mipmap` | Indicates if the device supports allocating 
+mipmap resources.
+|`aspect::ext_oneapi_mipmap_anisotropy` | Indicates if the device supports 
+sampling mipmap images with anisotropic filtering
+|`aspect::ext_oneapi_mipmap_level_reference` | Indicates if the device supports 
+using images created from individual mipmap levels
+|======================
+
+The device information descriptors for these queries are:
+
+[frame="none",options="header"]
+|======================
+|Device descriptor |Return type |Description
+|`ext::oneapi::experimental::info::device::mipmap_max_anisotropy` |`float` |
+Return the maximum anisotropic ratio supported by the device
+|======================
+
+=== Allocation of mipmapped images
+
+Mipmaps are allocated in a similar manner to standard images, however, mipmaps 
+do not support USM backed memory.
+
+Mipmap memory is allocated through `alloc_image_mem`. The user should populate 
+the `image_descriptor` with the image type of `image_type::mipmap`, and provide 
+the number of mipmaps levels they wish to allocate. The value of `num_levels` 
+must be greater than `1`.
+
+Mipmap memory allocated this way requires the user to free that memory after all 
+operations using the memory are completed and no more operations operating on 
+the memory will be scheduled. This is done using `free_image_mem`, passing 
+`image_type::mipmap`. Importantly, individual levels of a mipmap must not be 
+freed before calling `free_image_mem`.
+
+The RAII class `image_mem` may also be used to perform allocation and 
+deallocation of mipmap device memory. The constructor and destructor act as a 
+wrapper for the functions `alloc_image_mem` and `free_image_mem` respectively.
+
+When the underlying memory of `image_mem` is a mipmap, 
+`get_mip_level_mem_handle` can be used to return an `image_mem_handle` to a 
+specific level of the mipmap. This can then be used to copy data to that 
+specific level or create an image based on that level. 
+
+=== Obtaining a handle to a mipmap image
+
+A handle to a mipmap image is acquired in the same way as a 
+`sampled_image_handle`. Mipmaps can only be sampled image types. We can create a 
+`sampled_image_handle` to the allocated mipmap through the `create_image` 
+functions which take a sampler.
+
+Below we describe an extension to the existing SYCL `sampler` object which 
+allows for the sampling of mipmaps with support for level of detail(LOD) and 
+anisotropic filtering. To sample a mipmap correctly, the extended attributes of 
+this sampler must be defined.
+
+Attempting to create an `unsampled_image_handle` to a mipmap will result in a 
+`sycl::exception` with error code `sycl::errc::runtime` being thrown.
+
+=== Extended sampler object
+
+To represent the different sampling capabilities that mipmaps provide as opposed 
+to standard sampled images, we propose to extend the core SYCL `sampler` 
+class.
```

Comment:
The `sampler` class was removed with SYCL 2020 so we cannot and should not extend it. We could extend `image_sampler` but I would prefer we instead define a new sampler class to avoid the need to add `ext_` to [all the new members](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_names_for_extensions_to_existing_classes_or_enumerations).

Reply:
Noted, we're working on removing the `sycl::sampler` extension and creating our own sampler struct. This will involve changes to our [PR 2/4](https://github.com/intel/llvm/pull/10112), hopefully removing the need for ABI-Break completely. We should have an update for the implementation and proposal soon.